### PR TITLE
Fixes #22988: Rewrite file manager to an elm application

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/22988

The goal of this pull request is to replace our angularjs file managerhttps://github.com/joni2back/angular-filemanager by one inspired by https://github.com/jakeactually/elm-file-manager made in elm.

For this first iteration, i have adapted the elm app to the old api, that is much more detailed than the one from the base file manager. 

I added a new api function to create a file, but it misses two functions for now that were there before, editing file content and editing permissions

The css will be improved in a next pr

One of the major changes is now that the file manager can create files move them and upload them to the shared folder, which was impossible before

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/FileManager/View.elm [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  , div [ class "fm-name" ] [ text file.name ]
+  ]
+
+renderThumb : String -> String -> String -> FileMeta -> Html Msg
+renderThumb thumbApi api dir file = if file.type_ == "dir"
+  then div [ class "fm-thumb" ] [ fileIcon ]
+  else renderFileThumb api thumbApi <| dir ++ file.name
+
+fileIcon : Html Msg
+fileIcon = svg [ attribute "height" "48", viewBox "0 0 24 24", attribute "width" "48", attribute "xmlns" "http://www.w3.org/2000/svg" ]
```
Comment (fanf) [inline webapp/sources/rudder/rudder-web/src/main/elm/sources/FileManager/View.elm:124]: cc @RaphaelGauthier - if one day you need to draw svg by hand in elm, it's own it's done

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-web/src/main/elm/sources/FileManager/View.elm:124]: You won't have to keep an svg (and i hope you will change it @RaphaelGauthier !!)

**Comment by fanf** [comment]
Added a PR to make config-repo path a parameter, else I wasn't able to test it https://github.com/VinceMacBuche/rudder/pull/27 

I get an error when I try to update a technique with a new ressource, it says "path" is missing in json request. 
image

Works great for shared file.

I'm a bit amazed by the responsivness of the file manager.

**Comment by VinceMacBuche** [comment]
It was a problem with yaml pr more than this one in fact (encoder in elm part)

**Comment by VinceMacBuche** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]
LGTM!

**Pull request merged**